pkg/resources/static_registry.go

```diff
@@ -45,6 +46,58 @@ func (r *StaticToolRegistry) RegisterTool(tool protocol.Tool, handler ToolHandle
 	return nil
 }
 
+// RegisterStructToolWithHandler registers a tool with a typed handler function
+func (r *StaticToolRegistry) RegisterStructToolWithHandler(name, description string, structType reflect.Type, handlerFunc interface{}) error {
```

Comment:
Is it possible to do this with generics rather than the reflect type? I think we should do this with a two type generic, an input and an output - we can use teh schema struct annotation for both, this is actually a new feature in mcp june 2025 version